addon/components/paper-icon.js

```diff
@@ -4,7 +4,37 @@ export default Ember.Component.extend({
   tagName: 'md-icon',
   classNames: ['paper-icon'],
   classNameBindings: ['iconClass'],
+
+  spin: false,
+  reverseSpin: false,
+
   iconClass: function(){
-    return 'ic-'+this.get('icon');
+    var iconClasses = 'ic-'+this.get('icon');
+
+    if(this.get('spin')){
+      iconClasses +=' md-spin';
+    } else if (this.get('reverseSpin')){
+      iconClasses +=' md-spin-reverse';
+    }
+
+    switch(this.get('size')){
+      case 'lg':
+        iconClasses += ' md-lg';
+        break;
+      case 2:
+        iconClasses += ' md-2x';
+        break;
+      case 3:
+        iconClasses += ' md-3x';
+        break;
+      case 4:
+        iconClasses += ' md-4x';
+        break;
+      case 5:
+        iconClasses += ' md-5x';
+        break;
+    }
+
+    return iconClasses;
   }.property('icon')
```

Comment:
computed property is only depending on `icon`.

I think it makes sense to add separate this logic into different computed properties like `sizeClass` and `spinClass` and then add them do the classNameBindings like `classNameBindings: ['iconClass', 'sizeClass', 'spinClass']`.